verilog/analysis/verilog_linter.cc

```diff
@@ -77,12 +83,226 @@ using verible::LintWaiver;
+}
+
+void ViolationFixer::HandleViolation(const verible::LintViolation& violation,
+                                     absl::string_view base,
+                                     absl::string_view path,
+                                     absl::string_view url,
+                                     absl::string_view rule_name) {
+  std::stringstream violation_message;
+  formatter_->FormatViolation(&violation_message, violation, base, path, url,
+                              rule_name);
+  (*stream_) << violation_message.str() << std::endl;
+
+  if (violation.autofixes.size() > 0) {
+    AnswerChoice answer;
+    for (;;) {
+      if (ultimate_answer_ != AnswerChoice::kUnknown) {
+        answer = ultimate_answer_;
+      } else if (rule_answers_.find(rule_name) != rule_answers_.end()) {
+        answer = rule_answers_.at(rule_name);
+      } else {
+        answer = answer_chooser_(violation, rule_name);
+      }
+
+      switch (answer) {
+        case AnswerChoice::kApplyAll:
+          ultimate_answer_ = AnswerChoice::kApply;
+          [[fallthrough]];
+        case AnswerChoice::kApplyAllForRule:
+          rule_answers_[rule_name] = AnswerChoice::kApply;
+          [[fallthrough]];
+        case AnswerChoice::kApply:
+          // Apply first available fix
+          if (!fix_->AddEdits(violation.autofixes[0].Edits())) {
+            VLOG(2) << "The fix conflicts with previously applied fixes, "
+                       "rejecting.";
+          } else {
+            (*stream_) << "(fixed)" << std::endl;
+          }
+          break;
+
+        case AnswerChoice::kRejectAll:
+          ultimate_answer_ = AnswerChoice::kReject;
+          [[fallthrough]];
+        case AnswerChoice::kRejectAllForRule:
+          rule_answers_[rule_name] = AnswerChoice::kReject;
+          [[fallthrough]];
+        case AnswerChoice::kReject:
+          return;
+
+        case AnswerChoice::kPrintFix:
+          // Print first available fix
+          PrintFix(*stream_, base, violation.autofixes[0]);
+          continue;
+        case AnswerChoice::kPrintAppliedFixes:
+          PrintFix(*stream_, base, *fix_);
+          continue;
+
+        default:
+          continue;
+      }
+
+      break;
+    }
+  }
+}
+
+ViolationFixer::AnswerChoice ViolationFixer::InteractiveAnswerChooser(
+    const verible::LintViolation& violation, absl::string_view rule_name) {
+  static absl::string_view help_message =
+      "y - apply fix\n"
+      "n - reject fix\n"
+      "a - apply this and all remaining fixes for violations of this rule\n"
+      "d - reject this and all remaining fixes for violations of this rule\n"
+      "A - apply this and all remaining fixes\n"
+      "D - reject this and all remaining fixes\n"
+      "p - show fix\n"
+      "P - show fixes applied in this file so far\n"
+      "? - print this help and prompt again\n";
+
+  for (;;) {
+    const char c = verible::ReadCharFromUser(
+        std::cin, std::cerr, (isatty(STDIN_FILENO) == 1),
```

Comment:
this now can be the `IsInteractiveTerminalSession()` call.

Reply:
Done